### DISCUSSION TITLE
Update metadata on recipes to support link preview on iOS

### DISCUSSION THREAD

**Discussion by greirson** [mealie-recipes/mealie#4485] [Feature Request]
### First Check

- [X] I used the GitHub search to find a similar requests and didn't find it.
- [X] Checked the [tasks tagged](https://github.com/mealie-recipes/mealie/issues?q=is%3Aissue+is%3Aopen+label%3Atask+) issues and verified my feature is not covered

### Please provide a concise description of the problem that would be addressed by this feature.

Currently when sending a link to a recipe via iOS messages there is a generic title and image in the preview, instead of the Recipe name and the image associated with the recipe.

![IMG_0004](https://github.com/user-attachments/assets/866aff70-9d35-4ed9-80f2-717d22a6052c)

### Please provide a concise description of the feature that would resolve your issue.

iOS messages leverages [Open Graph metadata](https://ogp.me/) to retrieve the preview data. It would be great to implement the Title and Image from the recipe as [described here](https://developer.apple.com/library/archive/technotes/tn2444/_index.html)  

### Please consider and list out some caveats or tradeoffs made in your design decision

May impact recipe scraping? I’m unsure of any other caveats.

### Additional Information

- [ ] If this is accepted I'm willing to submit a PR to provide this feature
- [ ] If this is accepted I'm willing to help maintain this feature
- [X] I'm willing to sponsor/pay a developer to do this work

**Comment by michael-genson** [comment]
Mealie does implement open graph metadata, specifically for link previews, but the recipe has to be public (otherwise the recipe data can't be scraped). Are your recipes publically accessible?

**Reply by greirson** [reply]
Yes the recipe and the household are set to public and its visible it while not logged in. But the metadata shows the generic site data not data from the recipe, even when logged in .
<img width="1353" alt="Screenshot 2024-10-31 at 22 48 19" src="https://github.com/user-attachments/assets/b3a3ddb9-18ec-4da8-b767-80cf98c61f61">

**Reply by michael-genson** [reply]
Hmm, maybe it's an issue with IOS? Can you try this link and see if you get a preview? I confirmed it works on Discord.

https://demo.mealie.io/g/home/r/crispy-prosciutto-lemon-pesto-pasta-with-broccoli-parmesan

**Reply by michael-genson** [reply]
![Screenshot_20241101_010413_Discord.jpg](https://github.com/user-attachments/assets/2c16478a-675a-4d27-a8e8-976236cba22c)

**Reply by greirson** [reply]
It gets the title but no image for that link in iOS Messages. I'm running 2.1.0, any idea what could block it? 

<img width="369" alt="image" src="https://github.com/user-attachments/assets/462febc2-d806-411a-ba1e-f829304a3fd3">

**Reply by michael-genson** [reply]
I'll have to look into why the image isn't working on iOS but working fine on Discord (and other apps). For why the text isn't working for your site I'm not sure. Are you serving with a valid cert over HTTPS? Not sure if that matters but it could.

Can you scrape your site with curl and see the metadata in the response? e.g. `curl https://your-site.com/g/home/r/your-recipe`

**Comment by elcid2012** [comment]
@michael-genson I’m also having the same issue in iOS. My website is also publicly accessible and all recipes are public, but I still do not have previews in iMessage.

I’m not certain this is specific to iOS but rather to the website itself. In my case I’ve mealie running behind a reverse proxy locally, and Cloudflare tunnels externally. 

So I ran a curl command via https://reqbin.com/curl on both my mealie instance and the demo instance. All of the metadata seems to be similar, but I did find something that might point to part of the problem in the response.

The og:image is returning https://demo.mealie.io on the demo site, but http://mealie.example.com:8080 via curl. Now, I’d say that that’s wrong, but if I browse to http://mealie.example.com:8080/path-to-image, it does indeed load the image in my browser.

But perhaps the way the iOS scraper works, it’s not successful in pulling the image. After erroring out on that, perhaps it gives up parsing the rest of the metadata and so we see a generic website link instead of pretty open graph data.

Just spitting out some ideas. I’d like to see if I can change the URL path to the proper string somehow… saving that experiment for the morning.
